scrapy/core/engine.py

```diff
@@ -23,11 +25,15 @@
 
 class Slot:
 
-    def __init__(self, start_requests, close_if_idle, nextcall, scheduler):
+    def __init__(self, start_requests, close_if_idle, nextcall, scheduler, new_queue_behavior=False):
```

Comment:
The new_queue_behavior may become not so new after some time :) What do you think about calling the new argument old_queue_behavior, or, even better, with some name which describes the behavior?

Reply:
I'm fine with renaming this flag to whatever someone proposes.

Reply:
Maybe something about "eager" or "fair"?